src/DotVVM.Framework/Controls/UpdateProgress.cs

```diff
@@ -21,6 +36,11 @@ protected override void AddAttributesToRender(IHtmlWriter writer, IDotvvmRequest
         {
             writer.AddKnockoutDataBind("dotvvm-UpdateProgress-Visible", "true");
 
+            if (Delay != 0)
```

Comment:
Then the browser minimum time ​(~4ms) will be used, I actually think that having the possibility to force min time delay by `-1` is a good feature. And IE 8 will crash earlier :P